pkg/helpers/apply/applierbuilder.go

```diff
@@ -76,3 +85,12 @@ func (a *ApplierBuilder) WithOwner(owner runtime.Object, blockOwnerDeletion, con
 	a.scheme = scheme
 	return a
 }
+
+func (a *ApplierBuilder) WithCache(cache resourceapply.ResourceCache) *ApplierBuilder {
+	a.cache = NewResourceCache()
```

Comment:
why newCache in withCache, given you have a cache as input of the func.

Reply:
1. To avoid the developer to have to call the NewCache from the ressourceapply package.
2. WithCache allows the developer to use another implementation of the cache if he doesn't want to use the one provided by resourceappply.

Reply:
so that means in this func you should do like `a.cache = cache`, right?

Reply:
oops